src/nodejsWorker.ts

```diff
@@ -19,6 +19,13 @@ function validateNodeVersion(version) {
         if (versionSplit.length != 3) {
             message = "Could not parse Node.js version: '" + version + "'";
             // Unsupported version note: Documentation about Node's stable versions here: https://github.com/nodejs/Release#release-plan and an explanation here: https://medium.com/swlh/understanding-how-node-releases-work-in-2018-6fd356816db4
+        } else if (process.env.AZURE_FUNCTIONS_ENVIRONMENT == 'Development' && major == 'v17') {
+            const warning =
+                'Node.js version used (' +
+                version +
+                ') is not officially supported. You may use it during local development, but your function will be deployed to a different Node.js version on Azure.' +
+                ' Refer to our documentation to see the Node.js versions supported by each version of Azure Functions: https://aka.ms/functions-node-versions';

```

Comment:
nit: maybe we can shorten this a bit

```suggestion
                ') is not officially supported. You may use it during local development, but must use an officially supported version on Azure: https://aka.ms/functions-node-versions';

```